packages/router-plugin/tests/router.plugin.spec.ts

```diff
@@ -133,45 +155,348 @@ describe('NgxsRouterPlugin', () => {
+      })
+      class TestModule {}
+
+      return TestModule;
+    }
+
+    it('should wait for resolvers to complete and dispatch the `RouterDataResolved` event', async () => {
+      // Arrange
+      destroyPlatformBeforeBootstrappingTheNewOne();
+
+      // Act
+      const { injector } = await platformBrowserDynamic().bootstrapModule(getTestModule());
+      const router: Router = injector.get(Router);
+      const store: Store = injector.get(Store);
+
+      // Assert
+      const dataFromTheOriginalRouter = router.routerState.snapshot.root.firstChild!.data;
+      expect(dataFromTheOriginalRouter).toEqual({ test });
+
+      const dataFromTheRouterState = store.selectSnapshot(RouterState.state)!.root.firstChild!
+        .data;
+      expect(dataFromTheOriginalRouter).toEqual(dataFromTheRouterState);
+
+      resetPlatformAfterBootstrapping();
+    });
+
+    it('should keep resolved data if the navigation was performed between the same component but with params', async () => {
+      // Arrange
+      destroyPlatformBeforeBootstrappingTheNewOne();
+
+      // Act
+      const { injector } = await platformBrowserDynamic().bootstrapModule(getTestModule());
+      const router: Router = injector.get(Router);
+      const store: Store = injector.get(Store);
+
+      await store
+        .dispatch(
+          new Navigate(
+            ['/'],
+            {
+              a: 10
+            },
+            {
+              queryParamsHandling: 'merge'
+            }
+          )
+        )
+        .toPromise();
+
+      await store
+        .dispatch(
+          new Navigate(
+            ['/'],
+            {
+              b: 20
+            },
+            {
+              queryParamsHandling: 'merge'
+            }
+          )
+        )
+        .toPromise();
+
+      // Assert
+      const dataFromTheOriginalRouter = router.routerState.snapshot.root.firstChild!.data;
+      expect(dataFromTheOriginalRouter).toEqual({ test });
+
+      const dataFromTheRouterState = store.selectSnapshot(RouterState.state)!.root.firstChild!
+        .data;
+      expect(dataFromTheOriginalRouter).toEqual(dataFromTheRouterState);
+
+      resetPlatformAfterBootstrapping();
+    });
+
+    it('should dispatch `RouterDataResolved` action', async () => {
+      // Arrange
+      destroyPlatformBeforeBootstrappingTheNewOne();
+
+      // Act
+      const { injector } = await platformBrowserDynamic().bootstrapModule(getTestModule());
+      const actions$: Actions = injector.get(Actions);
+      const store: Store = injector.get(Store);
+
+      // The very first `ResolveEnd` event is triggered during root module bootstrapping
+      actions$
+        .pipe(
+          ofActionSuccessful(RouterDataResolved),
+          first()
+        )
+        .subscribe(({ routerState }: RouterDataResolved) => {
+          // Assert
+          const dataFromTheEvent = routerState.root.firstChild!.data;
+          expect(dataFromTheEvent).toEqual({ test });
```

Comment:
If we just capture the data in the subscribe into a variable then we can move this `expect` out of the subscribe and into the main body of the test. This makes for a test that can detect failure more reliably.